fontforge/fvfonts.c

```diff
@@ -1280,35 +1283,138 @@ static RefChar *InterpRefs(RefChar *base, RefChar *other, real amount, SplineCha
+        prefix = "";
+    } else {
+        prefix = smprintf(_("Glyph %s: "), glyphname);
+    }
+
+    if ( brc != NULL || orc != NULL ) {
+        RefChar *temp = InterpRefs(brc, orc, 0.5, glyphname, true);
+        ret = temp != NULL;
+    }
+
+    if (bmcc != omcc) {
+        LogError(_("%sIncompatible contour counts: %d vs %d."), prefix, bmcc, omcc);
+        ret = false;
+    }
+
+    SplinePoint* sp;
+    int contour_count = 1;
+    for ( bss = base, oss = other; bss!=NULL; bss=bss->next, oss=oss->next, contour_count++ ) {
+        if (!ret) break;
+        int bicw = SplinePointListIsClockwise(bss), oicw = SplinePointListIsClockwise(oss);
+
+        if (bicw == -1 || oicw == -1)
+            TRACE(_("%sInterpolationSanity: Assuming contour %d of undefined direction to be sane\n"), prefix, contour_count);
+        else if (bicw != oicw) {
+            LogError(_("%sIncompatible contour directionality (contour %d)"), prefix, contour_count);
+            ret = false;
+        }
+
+        if ( (bss->first->prev == NULL) != (oss->first->prev == NULL) ) {
+            char* openness = bss->first->prev == NULL ? _("closed") : _("open");
+            LogError(_("%sBase spline (contour %d) should be %s but it's not"), prefix, contour_count, openness);
+            ret = false;
+        }
+
+        int segment_count = 1;
+        for ( SplinePoint *bsp=bss->first, *osp=oss->first; ; segment_count++ ) {
+            bool bcurve = ( !bsp->nonextcp || (bsp->next!=NULL && !bsp->next->to->noprevcp) );
+            bool ocurve = ( !osp->nonextcp || (osp->next!=NULL && !osp->next->to->noprevcp) );
+
+            if ( bcurve != ocurve ) {
+                char* wrongpttype = bcurve ? _("base") : _("other");
+                LogError(_("%sSegment %d in the %s spline is a curve when it should be a line"), prefix, segment_count, wrongpttype);
+                ret = false;
+            }
+
+            bool bnonextpt = ( bsp->next==NULL || bsp->next->to==bss->first );
+            bool ononextpt = ( osp->next==NULL || osp->next->to==oss->first );
+
+            if ( bnonextpt && ononextpt ) 
+                  break;
+
+            if ( bnonextpt != ononextpt ) {
+                char* problemspline = bnonextpt ? _("base") : _("other");
+                int extras = 0;
+
+                if ( ononextpt ) {
+                    for ( ; bsp->next!=NULL && bsp->next->to==bss->first; bsp=bsp->next->to, extras++ );
+                } else if ( bsp->next==NULL || bsp==bss->first ) {
+                    for ( ; osp->next!=NULL && osp->next->to!=bss->first; osp=osp->next->to, extras++ );
+                }
+
+                LogError(_("%sThe %s spline has %d point(s) too many after segment %d"),
+                         prefix, problemspline, extras, segment_count);
+                ret = false; break;
+            }
+
+            bsp = bsp->next->to;
+            osp = osp->next->to;
+        }
+    }
+
+    if (prefix[0] != '\0') free(prefix);
```

Comment:
This works but is a little opaque. Seems better just to use the same logic as at allocation time (i.e. `glyphname!=NULL`)